src/libs/actions/Policy.ts

```diff
@@ -4005,17 +4005,7 @@ function navigateWhenEnableFeature(policyID: string, featureRoute: Route) {
         return;
     }
 
-    /**
-     * The app needs to set a navigation action to the microtask queue, it guarantees to execute Onyx.update first, then the navigation action.
-     * More details - https://github.com/Expensify/App/issues/37785#issuecomment-1989056726.
-     */
-    new Promise<void>((resolve) => {
-        resolve();
-    }).then(() => {
-        requestAnimationFrame(() => {
-            Navigation.navigate(featureRoute);
-        });
-    });
+    Navigation.setNavigationActionToMicrotaskQueue(() => Navigation.navigate(featureRoute));
```

Comment:
NAB but as we made changes here, let us add a test case to avoid possible regression.